crm-plugin/controllers/activity.js

```diff
@@ -0,0 +1,147 @@
+"use strict";
+
+/**
+ * crm-plugin.js controller
+ *
+ * @description: A set of functions called "actions" of the `crm-plugin` plugin.
+ */
+const { sanitizeEntity } = require("strapi-utils");
+module.exports = {
+  /**
+   * Default action.
+   *
+   * @return {Object}
+   */
+
+  index: async (ctx) => {
+    // Add your own logic here.
+    // Send 200 `ok`
+    ctx.send({
+      message: "ok",
+    });
+  },
+
+  find: async (ctx) => {
+    let activity;
+    try {
+      if (ctx.query._q) {
+        activity = await strapi
+          .query("activity", "crm-plugin")
+          .search(ctx.query);
+      } else {
+        activity = await strapi.query("activity", "crm-plugin").find(ctx.query);
+      }
+      return activity.map((entity) =>
+        sanitizeEntity(entity, {
+          model: strapi.plugins["crm-plugin"].models["activity"],
+        })
+      );
+    } catch (error) {
+      console.error(error);
+      return ctx.badRequest(null, error.message);
+    }
+  },
+
+  findOne: async (ctx) => {
+    const { id } = ctx.params;
+    try {
+      const entity = await strapi
+        .query("activity", "crm-plugin")
+        .findOne({ id });
+      return sanitizeEntity(entity, {
+        model: strapi.plugins["crm-plugin"].models["activity"],
+      });
+    } catch (error) {
+      console.error(error);
+      return ctx.badRequest(null, error.message);
+    }
+  },
+
+  create: async (ctx) => {
+    let actType;
+    let activityassignee;
+    let activity;
+    const demoParams = {
+      title: "name",
+      start_datetime: "2020-04-14T06:30:00.000Z",
+      end_datetime: "2020-04-10T06:30:00.000Z",
+    };
+    const reqVal = ["title"];
+    const result = strapi.plugins["crm-plugin"].services.utils.checkParams(
+      ctx.request.body,
+      reqVal
+    );
+    // console.log("result------", result);
+    if (result.error == "false") {
+      try {
+        if (ctx.params.id) {
+          const { id } = ctx.params;
+          activity = await strapi
+            .query("activity", "crm-plugin")
+            .update({ id }, ctx.request.body);
+          if (ctx.request.body.contact) {
+            let activityDetail = {
+              activity: activity.id,
+              contact: ctx.request.body.contact,
+            };
+            activityassignee = await strapi
+              .query("activityassignee", "crm-plugin")
+              .update({ activity: ctx.params.id }, activityDetail);
+          }
+          return sanitizeEntity(activity, {
+            model: strapi.plugins["crm-plugin"].models["activity"],
+          });
+        } else {
+          activity = await strapi
+            .query("activity", "crm-plugin")
+            .create(ctx.request.body);
+          if (ctx.request.body.contact) {
+            let activityDetail = {
+              activity: activity.id,
+              contact: ctx.request.body.contact,
+            };
+            activityassignee = await strapi
```

Comment:
order wrong. contact and activityassignee has to be updated first and then activity

Reply:
done